sql/src/main/java/io/crate/planner/statement/CopyToPlan.java

```diff
@@ -0,0 +1,299 @@
+import io.crate.analyze.PartitionPropertiesAnalyzer;
+import io.crate.analyze.QueriedSelectRelation;
+import io.crate.analyze.QuerySpec;
+import io.crate.analyze.SymbolEvaluator;
+import io.crate.analyze.WhereClause;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.common.collections.Lists2;
+import io.crate.data.Row;
+import io.crate.data.RowConsumer;
+import io.crate.exceptions.PartitionUnknownException;
+import io.crate.exceptions.UnsupportedFeatureException;
+import io.crate.execution.dsl.phases.NodeOperationTree;
+import io.crate.execution.dsl.projection.MergeCountProjection;
+import io.crate.execution.dsl.projection.WriterProjection;
+import io.crate.execution.dsl.projection.builder.ProjectionBuilder;
+import io.crate.execution.engine.NodeOperationTreeGenerator;
+import io.crate.expression.symbol.Literal;
+import io.crate.expression.symbol.Symbol;
+import io.crate.expression.symbol.Symbols;
+import io.crate.expression.symbol.format.SymbolPrinter;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.DocReferences;
+import io.crate.metadata.Functions;
+import io.crate.metadata.GeneratedReference;
+import io.crate.metadata.Reference;
+import io.crate.metadata.doc.DocSysColumns;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.planner.DependencyCarrier;
+import io.crate.planner.ExecutionPlan;
+import io.crate.planner.Merge;
+import io.crate.planner.Plan;
+import io.crate.planner.PlannerContext;
+import io.crate.planner.SubqueryPlanner;
+import io.crate.planner.consumer.FetchMode;
+import io.crate.planner.operators.LogicalPlan;
+import io.crate.planner.operators.LogicalPlanner;
+import io.crate.planner.operators.SubQueryResults;
+import io.crate.sql.tree.Assignment;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import static io.crate.analyze.CopyStatementSettings.COMPRESSION_SETTING;
+import static io.crate.analyze.CopyStatementSettings.OUTPUT_FORMAT_SETTING;
+import static io.crate.analyze.CopyStatementSettings.OUTPUT_SETTINGS;
+import static io.crate.analyze.CopyStatementSettings.settingAsEnum;
+import static io.crate.analyze.GenericPropertiesConverter.genericPropertiesToSettings;
+
+public final class CopyToPlan implements Plan {
+
+    private final AnalyzedCopyTo copyTo;
+    private final LogicalPlanner logicalPlanner;
+    private final SubqueryPlanner subqueryPlanner;
+
+    public CopyToPlan(AnalyzedCopyTo copyTo,
+                      LogicalPlanner logicalPlanner,
+                      SubqueryPlanner subqueryPlanner) {
+        this.copyTo = copyTo;
+        this.logicalPlanner = logicalPlanner;
+        this.subqueryPlanner = subqueryPlanner;
+    }
+
+    AnalyzedCopyTo copyTo() {
```

Comment:
please add the `VisibleForTesting` annotation to the getters, I was wondering first where these are used..